pkg/client/patch.go

```diff
@@ -148,7 +164,21 @@ func MergeFromWithOptions(obj runtime.Object, opts ...MergeFromOption) Patch {
 	for _, opt := range opts {
 		opt.ApplyToMergeFrom(options)
 	}
-	return &mergeFromPatch{from: obj, opts: *options}
+	return &mergeFromPatch{patchType: types.MergePatchType, createPatch: createMergePatch, from: obj, opts: *options}
+}
+
+// StrategicMergeFrom creates a Patch that patches using the strategic-merge-patch strategy with the given object as base.
+func StrategicMergeFrom(obj runtime.Object) Patch {
+	return &mergeFromPatch{patchType: types.StrategicMergePatchType, createPatch: createStrategicMergePatch, from: obj}
+}
+
+// StrategicMergeFromWithOptions creates a Patch that patches using the strategic-merge-patch strategy with the given object as base.
+func StrategicMergeFromWithOptions(obj runtime.Object, opts ...MergeFromOption) Patch {
```

Comment:
Is it appropriate to use `MergeFromOption` here? My concern is that `merge-patch` and `strategic-merge-patch` could diverge in the options they take in the future.

Reply:
I was contemplating about the same, glad you bring it up here.

The reason I chose this way, is that if options specific to either patch type will be added in the future, we could simply add another interface `StrategicMergeFromOption`, implement both interfaces in the options that we already have and just switch the parameter type in `StrategicMergeFrom`. This way, folks using the options wouldn't get any compile tim error when upgrading.
And as it's not required to differentiate right now, I chose to reuse `MergeFromOption` for simplicity's sake.

Let me know, if you have any concerns with this approach.